### PR TITLE
asset cache + better minimaps

### PR DESCRIPTION
**Pull request by LetterN**
## About The Pull Request
Ports asset cache from tg.

## Why It's Good For The Game
No need to download everything when reconnecting

## Changelog
:cl:
add: Asset cache from tg
tweak: Made the map viewer not look bad
fix: Admin matrix right-bracket
/:cl:

### CODE REVIEW

File: code/controllers/subsystem/minimaps.dm [hunk 1]
```diff
@@ -1,8 +1,8 @@
 SUBSYSTEM_DEF(minimaps)
 	name = "Minimaps"
 	flags = SS_NO_FIRE
-	var/list/station_minimaps
-	var/datum/minimap_group/station_minimap
+	var/list/station_minimaps = list()
+	var/datum/minimap_group/station_minimap = null
```
Comment (Ghommie) [inline code/controllers/subsystem/minimaps.dm:5]: Not needed to state it's null, but whatever.

File: code/modules/asset_cache/asset_cache_client.dm [hunk 2]
```diff
@@ -0,0 +1,58 @@
+
+/client
+	var/list/sent_assets = list() // List of all asset filenames sent to this client by the asset cache, along with their assoicated md5s
+	var/list/completed_asset_jobs = list() /// List of all completed blocking send jobs awaiting acknowledgement by send_asset
+
+	var/last_asset_job = 0 /// Last asset send job id.
+	var/last_completed_asset_job = 0
```
Comment (Ghommie) [inline code/modules/asset_cache/asset_cache_client.dm:7]: Move these variables to the client_defines file.

File: code/modules/mapping/minimaps.dm [hunk 3]
```diff
@@ -60,61 +72,79 @@
...
+	var/list/minimaps = list()
 	var/static/next_id = 0
 	var/id
 	var/name
 
 /datum/minimap_group/New(list/maps, name)
 	id = ++next_id
 	src.name = name
-	minimaps = maps || list()
+	minimaps = maps
```
Comment (Ghommie) [inline code/modules/mapping/minimaps.dm]:
```suggestion
	if(maps)
		minimaps = maps
```

**Review by Ghommie — CHANGES_REQUESTED** [review]

File: code/modules/asset_cache/asset_cache_client.dm (hunk 2 above)
Comment (LetterN) [reply code/modules/asset_cache/asset_cache_client.dm:7]: but this is cache specific

Comment (Ghommie) [reply code/modules/asset_cache/asset_cache_client.dm:7]: Thing is this sort of things fucks up the DM langserv for VSC from what I heard.

Comment (LetterN) [reply code/modules/asset_cache/asset_cache_client.dm:7]: only if you're specifically finding `/client`, but the vars are fine. (langserv is less dumb now)

**Review by Ghommie — APPROVED** [review]

**Pull request merged**